### PR TITLE
Add tests to check for pelican 4

### PR DESCRIPTION
**Pull request by ashwinvis**
Redo #239 for issue #174

### CODE REVIEW

**Comment by talha131** [comment]
@iranzo please review this PR.

@ashwinvis I don't quite follow. What's the end result of this PR? Will you please walk me through it?

travis will run a job on next branch. Your PR has two `TEST_CMD`. Which one will run? 

Then then `SITEURL` is same as the one we use for hosting the documentation. What will be its affect?

So, I am confused here. Kindly help me understand this PR.

File: .travis.yml [hunk 1]
```diff
@@ -7,6 +7,10 @@ sudo: required
 python:
 - '3.5'
 
+env:
+- TEST_CMD="pelican content/ -o output/"
```
Comment (iranzo) [inline .travis.yml]: Redundant with below line

**Comment by iranzo** [comment]
As with Talha's comment, I don't see what else are we testing here vs what we do have now (by using the pelicanconf)

Would you please ellaborate what are we testing here that we don't have in current test?

Thanks  a lot!
Pablo

**Comment by iranzo** [comment]
Would it be also probably good to try putting whatever is needed in the pelicanconf instead of additional files (to help further contributors) unless there are specifics that should be outside

(pending on explanation to see what I'm missing)

**Comment by iranzo** [comment]
The unit testing doesn't actually render using the publishing configuration as we just want the website to build, the actual publication only happens at the documentation website Unit Testing which is the one in charge of refreshing https://pelican-elegant.github.io.

In Pelican conf, site URL is defined as "/", only thing will be the  missing 'FEED_ALL_ATOM' variables that are set to 'None'.

So that's ok for me to merge it, just in case, as we're using just one pelican command, remove the duplicated TEST_CMD or just move the command to use the publishconf and leave the full command

Would that be ok @ashwinvis ?

**Comment by ashwinvis** [comment]
Usually the default configuration in `pelicanconf.py` is `SITEURL = ""; RELATIVE_URLS = True` and vice versa in `publishconf.py`. If you (@iranzo) think this is unnecessary, feel free to combine it back into a single test.

Personally I think separate unittest are good practice to allow for debugging. As you can see the bug I mentioned in issue #174 only occurs if you use publish configuration

**Comment by iranzo** [comment]
It's ok, that's why I merged it already :-)

**Pull request merged**